plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -175,12 +180,31 @@ public void testV2TableWithEqualityDelete()
         String tableName = "test_v2_equality_delete" + randomTableSuffix();
         assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.tiny.nation", 25);
         Table icebergTable = updateTableToV2(tableName);
-        writeEqualityDeleteToNationTable(icebergTable);
+        writeEqualityDeleteToNationTable(icebergTable, Optional.of(icebergTable.spec()));
         assertQuery("SELECT * FROM " + tableName, "SELECT * FROM nation WHERE regionkey != 1");
         // natiokey is before the equality delete column in the table schema, comment is after
         assertQuery("SELECT nationkey, comment FROM " + tableName, "SELECT nationkey, comment FROM nation WHERE regionkey != 1");
     }
 
+    @Test
+    public void testOptmizingV2TableWithEqualityDelete()
+            throws Exception
+    {
+        String tableName = "test_v2_equality_delete" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " WITH (partitioning = ARRAY['regionkey']) AS SELECT * FROM tpch.tiny.nation", 25);
+        Table icebergTable = updateTableToV2(tableName);
+        Assertions.assertThat(icebergTable.currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("0");
+        writeEqualityDeleteToNationTable(icebergTable, Optional.of(icebergTable.spec()));
+        List<String> initialActiveFiles = getActiveFiles(tableName);
+        query("ALTER TABLE " + tableName + " EXECUTE OPTIMIZE");
+        assertQuery("SELECT * FROM " + tableName, "SELECT * FROM nation WHERE regionkey != 1");
+        // natiokey is before the equality delete column in the table schema, comment is after
+        assertQuery("SELECT nationkey, comment FROM " + tableName, "SELECT nationkey, comment FROM nation WHERE regionkey != 1");
+        Assertions.assertThat(icebergTable.currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("1");
```

Comment:
Gotta reload the table here, either from scratch
```suggestion
        Assertions.assertThat(loadTable().currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("1");
```

or call `icebergTable.refresh()`